pydantic/_internal/_utils.py

```diff
@@ -387,3 +390,35 @@ def __getitem__(self, key: str, /) -> Any:
 
         def __contains__(self, key: str, /) -> bool:
             return self.wrapped.__contains__(key)
+
+
+_ModelT = TypeVar('_ModelT', bound='BaseModel')
+_R = TypeVar('_R')
+
+
+class deprecatedinstanceproperty(Generic[_ModelT, _R]):
```

Comment:
```suggestion
class deprecated_instance_property(Generic[_ModelT, _R]):
```

maybe? easier to read